nvaccess/nvda
# Replace internal controlTypes.ROLE_* usages with controlTypes.Role.*

### Link to issue number:

Part of #12549 

### Summary of the issue:

`controlTypes.ROLE_*` are deprecated, to be replaced by `controlTypes.Role.*`

### Description of how this pull request fixes the issue:

* replaces all instances of `controlTypes.ROLE_` with `controlTypes.Role.` across .py files
* search and manually replace final ROLE_ constant usages 
     * Searched using the regex `[^\.]\bROLE_`, which finds ROLE_ instances
where:
          * it is not of the form `<Foo>.ROLE_ `(as oleacc.ROLE_* exists)
          * ROLE_ is the prefix of the symbol (as IA2_ROLE_* exists)

### Testing strategy:

Manual testing
* Confirm the above steps will not introduce issues, and cover all cases.
* Reproduce those steps and perform a diff with this branch

Existing unit tests exist for controlTypes, new unit tests are written to ensure displayString works

Existing system tests will work as a basic smoke test for this core feature

### Known issues with pull request:

The lint check is likely to fail

### Change log entries:

None needed

### Code Review Checklist:



  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
